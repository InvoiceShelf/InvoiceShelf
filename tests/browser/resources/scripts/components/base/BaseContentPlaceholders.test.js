import { h } from 'vue'
import { expect, test } from 'vitest'
import { render } from 'vitest-browser-vue'
import BaseContentPlaceholders from '@/scripts/components/base/BaseContentPlaceholders.vue'
import BaseContentPlaceholdersText from '@/scripts/components/base/BaseContentPlaceholdersText.vue'
import BaseContentPlaceholdersBox from '@/scripts/components/base/BaseContentPlaceholdersBox.vue'

test('BaseContentPlaceholders animated with BaseContentPlaceholdersText', async () => {
  const { getByTestId } = render(BaseContentPlaceholders, {
      props: { animated: true, 'data-testid': 'base-placeholder', 'class': 'flex flex-col gap-y-4' },
      slots: {
        default: () => [
          h(
            BaseContentPlaceholdersText,
            {
              'data-testid': 'text-placeholder'
            }
          ),
          h(
            BaseContentPlaceholdersBox,
            {
              rounded: true,
              'class': 'w-full h-24'
            }
          ),
          h(
            BaseContentPlaceholdersBox,
            {
              circle: true,
              'class': 'w-5 h-5'
            }
          )
        ],
      }
  })
  const basePlaceholder = getByTestId('base-placeholder')
  await expect.element(basePlaceholder).toBeInTheDocument()
  await expect.element(basePlaceholder).toBeVisible()

  await expect.element(basePlaceholder).toHaveClass('base-content-placeholders-is-animated')

  expect(basePlaceholder).toMatchSnapshot()
})

