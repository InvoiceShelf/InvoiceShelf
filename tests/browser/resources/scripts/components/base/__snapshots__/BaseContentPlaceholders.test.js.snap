// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseContentPlaceholders animated with BaseContentPlaceholdersText 1`] = `
PlaywrightLocator {
  "_container": <body>
    
    
    
  


    <div>
      <div
        class="base-content-placeholders base-content-placeholders-is-animated flex flex-col gap-y-4"
        data-testid="base-placeholder"
      >
        
        <div
          class="base-content-placeholders-text"
          data-testid="text-placeholder"
        >
          
          <div
            class="w-full h-full base-content-placeholders-text__line flex flex-col"
          />
          <div
            class="w-full h-full base-content-placeholders-text__line flex flex-col"
          />
          <div
            class="w-full h-full base-content-placeholders-text__line flex flex-col"
          />
          <div
            class="w-full h-full base-content-placeholders-text__line flex flex-col"
          />
          
        </div>
        <div
          class="base-content-placeholders-box base-content-placeholders-is-rounded w-full h-24"
        />
        <div
          class="base-content-placeholders-box base-content-circle w-5 h-5"
        />
        
      </div>
    </div>
  </body>,
  "_parsedSelector": {
    "capture": undefined,
    "parts": [
      {
        "body": [
          {
            "simples": [
              {
                "combinator": "",
                "selector": {
                  "css": "body",
                  "functions": [],
                },
              },
            ],
          },
        ],
        "name": "css",
        "source": "body",
      },
      {
        "body": "[data-testid="base-placeholder"s]",
        "name": "internal:testid",
        "source": "[data-testid="base-placeholder"s]",
      },
    ],
  },
  "_pwSelector": undefined,
  "selector": "body >> internal:testid=[data-testid="base-placeholder"s]",
}
`;
